pennylane/compiler/qjit_api.py

```diff
@@ -568,3 +605,26 @@ def __call__(self, *init_state):
             args = fn_res if len(args) > 1 else (fn_res,) if len(args) == 1 else ()
 
         return fn_res
+
+    def _call_capture_enabled(self, *init_state):
+
+        import jax  # pylint: disable=import-outside-toplevel
+
+        for_loop_prim = _get_for_loop_qfunc_prim()
+
+        jaxpr_body_fn = jax.make_jaxpr(self.body_fn)
```

Comment:
So this is a hybrid between the on-the-fly and staged processing?

https://jax.readthedocs.io/en/latest/autodidax.html#on-the-fly-final-style-and-staged-initial-style-processing

It seems like we would neither get the more versatile nature of on-the-fly processing or the straight forward, ahead of time information of staged processing.

I'd be fine with considering use of on-the-fly processing for the control flow primitives, but I'm curious why staged processing doesn't work in this situation.

Reply:
Considering how `qml.for_loop` works and the corresponding syntax, it seems to me that the most natural approach is to call `jax.make_jaxpr` 'capturing' the body function and then evaluate it with `eval_jaxpr` for every possible value of the index since the index is always the first argument of the body function (I honestly find the syntax of `qml.for_loop` a little counter-intuitive).

However, I am pretty agnostic on the implementation as soon as it does the job, and maybe there are better ways. What would you suggest?